### ISSUE TITLE
Add fee handling for Stacking

### ISSUE THREAD

**Issue by markmhx** [blockstack/stacks-wallet#279]
Currently the designs for the new Stacks Wallet don't represent the fees involved with Stacking in any way.

We should modify them to:

1. Show the user upfront how just how much they'll spend in STX fees to participate in Stacking
2. Provide accurate guidance upfront as to just how much they'll need in both minimum principal (e.g. 100,000 STX) _and_ fees to participate in Stacking, so they don't buy or transfer just the minimum principal into their wallet and be unable to Stack without getting more
3. Toggle the visibility of the "Get started" path into Stacking based on the principal plus fees total, not just principal.

Note that we'll need to support the above in the context of variable fee amounts, which can change theoretically by the minute or at least each session.

**Comment by jasperjansz** [comment]
Here's my proposal: https://www.figma.com/file/VcXbxT1EwXvE1nUlOiBEES/Wallet?node-id=498%3A21891

1. Show the fees in the summary card while you're setting up Stacking
2. My proposal would be to take the principal (let's say it's 99,247 STX), add a 1% buffer (99,247*1.01 = 100,239.47 STX), and round up to the nearest hundred to get a clean number (100,239.47 -> 100,300 STX). This doesn't change the design other than changing the number displayed here: https://www.figma.com/file/VcXbxT1EwXvE1nUlOiBEES/Wallet?node-id=442%3A1839
3. Yep. This won't change the design, just the number displayed.

To account for variable fees I propose we set the buffer percentage high enough. I presume 1% is enough but I'm not sure. I mentioned 3% during the meeting and people deemed that to be too high.

**Comment by kyranjamie** [comment]
Agreed. Think 1% is still way beyond what we need to consider a safe buffer, but we'll change the value accordingly.

**Comment by markmhx** [comment]
@hstove Perhaps you have some insight here on how @kyranjamie can get a more reliable fee estimation for contract-based transaction signing.
